server/controllers/availability.js

```diff
@@ -0,0 +1,139 @@
+const PetSitter = require("../models/PetSitter")
+const Schedule = require("../models/Schedule");
+const asyncHandler = require("express-async-handler");
+const Availability = require("../models/Availability");
+
+// @route POST /availability/schedule/:scheduleId
+// @desc create or update a schedule
+// @access Private
+exports.createSchedule = asyncHandler(async (req, res, next) => {
+  const petsitter = await PetSitter.findOne(req.user);
+  const availability = await Availability.findById(petsitter.availabilityId);
+  const scheduleIds = await availability.scheduleIds
+  const { name, MondayFrom, MondayTo, TuesdayFrom, TuesdayTo, WednesdayFrom, WednesdayTo, ThursdayFrom, ThursdayTo, FridayFrom, FridayTo, SaturdayFrom, SaturdayTo, SundayFrom, SundayTo } = req.body;
+  
+  const newPeriods = {
+    monday: {
+      startTime: MondayFrom,
+      endTime: MondayTo,
+    },
+    tuesday: {
+      startTime: TuesdayFrom,
+      endTime: TuesdayTo,
+    },
+    wednesday: {
+      startTime: WednesdayFrom,
+      endTime: WednesdayTo,
+    },
+    thursday: {
+      startTime: ThursdayFrom,
+      endTime: ThursdayTo,
+    },
+    friday: {
+      startTime: FridayFrom,
+      endTime: FridayTo,
+    },
+    saturday: {
+      startTime: SaturdayFrom,
+      endTime: SaturdayTo,
+    },
+    sunday: {
+      startTime: SundayTo,
+      endTime: SundayFrom,
+    },
+  };
+
+  if (scheduleIds.includes(req.params.scheduleId)) {
+    schedule = await Schedule.findById(req.params.scheduleId)
+    schedule.name = name;
+    schedule.periods = newPeriods;
+    await schedule.save()
+
```

Comment:
Semantically, this should only ever create a new schedule not edit an existing one. We should use PUT for that in a different route.

Reply:
updated so that we have two separate routes now: 
- one for new schedules: `/availability/schedule/new`
- one for updating existing schedules: `/availability/schedule/:scheduleId`